README.md

```diff
@@ -1,11 +1,74 @@
+The Rancher Webhook is one such controller that runs only in the local cluster.
+
+A typical admission controller, such as the Webhook, works as a web-server that receives requests forwarded
+by the main API server. If the Webhook runs its custom validation logic for the resource in question
+and rejects it because the object's structure violates some custom rules, then the whole request to the API server
+is also rejected.
+
+A significant part of the low-level functionality of the Webhook's web-server is implemented
+in [Wrangler](https://github.com/rancher/wrangler). Wrangler provides code for error handling, top-level routing,
+object decoding and response encoding.
+
+The Go web-server itself is configured by [dynamiclistener](https://github.com/rancher/dynamiclistener).
+It handles TLS certificates and the management of associated Secrets for secure communication of other Rancher components with the Webhook.
+
+## Webhook
+The Webhook focuses on the resource matching rules and admission (or rejection) logic for each relevant
+object that needs custom validation. Each object that the Webhook inspects has an `Admit` method which takes a webhook request and response. 
+Its main job is to extract the object from the request,
+perform custom validation or mutation, and modify the response as needed, thus signaling if validation succeeded
+or failed. When the Webhook is done, it returns control to Wrangler which then handles the sending of the verdict
+of the admission controller.
+
+```Admit(webhookResponse *Response, webhookRequest *Request) error```
+
+Note that each object's Admit method returns an error.
+This error is not to signify that an object violates some validation rules, but rather to signal that
+the validation process itself has encountered an error. If an admission controller has deemed a resource
+invalid for persistence, it means it itself has succeeded.
+
+Any admission controller, as an app, consists of two main things:
+1. The configuration which describes the resources and actions for which the webhook is active. This
+   configuration also references the Kubernetes service which directs traffic to the actual web-server
+   (this Webhook project) that does the work. The configuration exists as ValidatingWebhookConfiguration
+   and MutatingWebhookConfiguration global resources.
+2. The actual web-server with all the validation and mutation logic found in each object's Admit method.
+
+Both the ValidatingWebhookConfiguration and MutatingWebhookConfiguration are fully created at
+webhook startup, not beforehand. Usually these key admission controller resources are deployed as manifests along with those
+for the web-server deployment and service, but Rancher is different here.
+
+All objects with custom validation logic exist in the `pkg/resources` package. For example, there is a
+`globalrole` package that has a type with an Admit method. Every time the local cluster's kube-api server receives a
```

Comment:
Can you link to this one. The bottom part of paragraph could use a link to the specific verb check too: https://github.com/rancher/webhook/blob/b9dc2209f175f0e052f51db7c09eb892cc514059/pkg/resources/validation/globalrole/globalrole.go#L42-L49